drivers/spi/spi_sy1xx.c

```diff
@@ -0,0 +1,521 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) 2024 sensry.io
+ */
+
+#define DT_DRV_COMPAT sensry_sy1xx_spi
+
+#define LOG_LEVEL CONFIG_SPI_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(spi_sy1xx);
+
+#include "spi_context.h"
+#include <errno.h>
+#include <zephyr/spinlock.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/util.h>
+#include <udma.h>
+#include <pad_ctrl.h>
+
+struct sy1xx_spi_config {
+	uint32_t base;
+	uint32_t inst;
+
+	uint8_t cpol;
+	uint8_t cpha;
+	uint8_t div;
+	uint8_t byteAlign;
+	uint8_t quad_spi;
+};
+
+#define DEVICE_MAX_BUFFER_SIZE (512)
+
+struct sy1xx_spi_data {
+	struct spi_context ctx;
+	struct k_spinlock lock;
+
+	uint8_t write[DEVICE_MAX_BUFFER_SIZE];
+	uint8_t read[DEVICE_MAX_BUFFER_SIZE];
+};
+
+typedef struct {
+	uint8_t cs;
+
+	uint32_t addr_width;
+	uint32_t addr;
+
+	uint16_t data_len;
+	uint8_t *data;
+
+} sy1xx_spiTransfer_t;
```

Comment:
just use something like `struct sy1xx_spi_xfer`/`sy1xx_spi_transfer`?

Reply:
done.

Reply:
can be marked resolved?